code/game/machinery/doors/airlock.dm

```diff
@@ -117,6 +118,27 @@
 		max_integrity = normal_integrity
 	if(damage_deflection == AIRLOCK_DAMAGE_DEFLECTION_N && security_level > AIRLOCK_SECURITY_METAL)
 		damage_deflection = AIRLOCK_DAMAGE_DEFLECTION_R
+	if(abandoned)
+		var/outcome = rand(1,100)
+		switch(outcome)
+			if(1 to 9)
+				var/turf/here = get_turf(src)
+				for(var/turf/closed/T in range(2, src))
+					here.ChangeTurf(T.type)
+					return INITIALIZE_HINT_QDEL
+				here.ChangeTurf(/turf/closed/wall)
+				return INITIALIZE_HINT_QDEL
+			if(9 to 11)
+				lights = FALSE
+				bolt()
```

Comment:
Just set `locked = TRUE`. The `bolt()` proc plays the sound and runs `update_icon()`. You obviously do not need to play the sound during mapload, and `update_icon()` is called for you at the end of this proc.

Reply:
I'd stick to the helper proc here without worrying about optimization.

Reply:
I changed my mind, change this to `locked = TRUE`